feat: Add spec.manager (#1958)
kyma-project/lifecycle-manager: docs/contributor/resources/03-moduletemplate.md

- documentation: The link to the documentation of the module.
- icons: A list of icons of the module, each with a name and link.

### **.spec.manager**

> The `manager` field provides information for identifying a module's resource that can be used as indicator for the installation readiness of the module. Typically, this is the manager Deployment of the module. In exceptional cases, it may also be another resource. If the resource is not namespaced, such as the module CR's `CRD`, the namespace can be optional.

Reviewer: ```suggestion
The `manager` field provides information for identifying a module's resource that can indicate the module's installation readiness. Typically, the manager is the module's `Deployment` resource. In exceptional cases, it may also be another resource. The **namespace** parameter is optional if the resource is not namespaced. For example,  if the resource is the module CR's `CustomResourceDefinition`.
```

> For example, if the module has a manager `Deployment` in the `kyma-system` namespace:

Reviewer: ```suggestion
In this example, the module's manager is the `Deployment` resource in the `kyma-system` namespace.
```

```yaml
spec:
  manager:
    group: apps
    version: v1
    kind: Deployment
    namespace: kyma-system
    name: [module manager name]
```

> For a module without a manager, such as [cloud manager](https://github.com/kyma-project/cloud-manager):

Reviewer: ```suggestion
In this example, the module doesn't have a manager and in the **manager** field you specifie the module's `CustomResourceDefinition`. For example, of the [Cloud Manager](https://github.com/kyma-project/cloud-manager) module:
```
I'm just not sure if it means that there is no manager or if the manager is the CRD?

> For a module without a manager, such as [cloud manager](https://github.com/kyma-project/cloud-manager):
> 
> ```yaml
> spec:
>   manager:
>     group: apiextensions.k8s.io
>     version: v1
>     kind: CustomResourceDefinition
>     name: [module CRD name]
> ```

Reviewer 2: ```suggestion
```

Thinking about it, maybe we should remove this example entirely. Just to not give the impression that this is something that should be done.

Author: if we need to have this use case, I think we should mention it, for documentation purpose, it also brings value,  implicitely as hidden rule which applied for certain modules could lead to more confusion.

However, as for whether it make sense of making CRD as the manager, we can open a dicussion on this. It might be better to leave this field optional, assuming that for modules without a manager, they are considered ready by default. Until a decision is made, I believe we should keep this use case as a valid example

> In this example, the module doesn't have a manager and in the **manager** field you specifie the module's `CustomResourceDefinition`. For example, of the [Cloud Manager](https://github.com/kyma-project/cloud-manager) module:

Reviewer: ```suggestion
In this example, the module's manager is the module's `CustomResourceDefinition` that does not require the **namespace** parameter.
```

- documentation: The link to the documentation of the module.
- icons: A list of icons of the module, each with a name and link.

### **.spec.manager**

> The `manager` field provides details about a module's resource that indicates its readiness. Typically, this is the
> `Deployment` managing the module. If the resource is not namespaced, such as the module CR's `CRD`, the namespace can be
> optional.

Reviewer 2: ```suggestion
The `manager` field provides information for identifying a module's resource that can be used as indicator for the installation readiness of the module. Typically, this is the manager Deployment of the module. In exceptional cases, it may also be another resource. If the resource is not namespaced, such as the module CR's `CRD`, the namespace can be optional.
```

For example, if the module has a manager `Deployment` in the `kyma-system` namespace:

```yaml
spec:
  manager:
    group: apps
    version: v1
    kind: Deployment
    namespace: kyma-system

>     name: [module manager]

Reviewer 2: ```suggestion
    name: [module manager name]
```

```

For a module without a manager, such as [cloud manager](https://github.com/kyma-project/cloud-manager):

```yaml
spec:
  manager:
    group: apiextensions.k8s.io
    version: v1
    kind: CustomResourceDefinition

>     name: [module CRD]

Reviewer 2: ```suggestion
    name: [module CRD name]
```

```

This field helps identify the module's manager, which is useful for determining whether a module is successfully

> installed, especially in manual installation scenarios.

Reviewer 2: I think this should be obvious from the above description. If we consider this to be helpful, I would at least move it up to the rest of the description.

Author: agree, removed
